src/Appwrite/Auth/OAuth2/Etsy.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace Appwrite\Auth\OAuth2;
+
+use Appwrite\Auth\OAuth2;
+
+class Etsy extends OAuth2
+{
+    /**
+     * @var string
+     */
+    private $endpoint = 'https://api.etsy.com/v3/public';
+
+    /**
+     * @var string
+     */
+    private $version = '2022-07-14';
+
+    /**
+     * @var array
+     */
+    protected $user = [];
+
+    /**
+     * @var array
+     */
+    protected $tokens = [];
+
+    /**
+     * @var array
+     */
+    protected array $scopes = [
+        "address_r",
+        "address_w",
+        "billing_r",
+        "cart_r",
+        "cart_w",
+        "email_r",
+        "favorites_r",
+        "favorites_w",
+        "feedback_r",
+        "listings_d",
+        "listings_r",
+        "listings_w",
+        "profile_r",
+        "profile_w",
+        "recommend_r",
+        "recommend_w",
+        "shops_r",
+        "shops_w",
+        "transactions_r",
+        "transactions_w",
```

Comment:
are all of those necessary to work?

Here should only be the bare minimum for it to work 👍🏻